### DISCUSSION TITLE
alias command doesn't respect project's team scope anymore

### DISCUSSION THREAD

**Discussion by simonhaenisch** [vercel/vercel#4390] [General] [answered]
Since I upgraded `now` to v19, it doesn't respect the project's team scope anymore when running the `alias` command. By project I mean that the folder that I run the command from is linked to the project.

I have a manual deployment npm script that does something like

```
URL=$(now --build-env ENV_NAME=prod) && now alias set ${URL} my-domain.com
```

It throws an error that no deployment with the URL could be found because it's trying to look for deployments in my default scope rather than the team scope that the project is linked to. This used to work. I need to add `--scope my-team-name` to the alias command to get it to work now.

BTW I tried changing everything to `vercel`, i. e. installed the CLI, changed the command names, and re-linked the project, but it's the same issue.

**Comment by simonhaenisch** [comment]
Hm I noticed that there's other commands (like `vercel ls`) that also don't get the team scope applied automatically anymore... was this an intentional change?

**Comment by styfle** [comment, marked as answer]
Yes, the project inheritance was changed in [19.0.0](https://github.com/zeit/now/releases/tag/vercel%4019.0.0) (see release notes).

We might have made a mistake with the `alias` subcommand. Let me verify with the team if that should use the local project or `vercel switch` scope, thanks!

*Update*: I confirmed this is the expected behavior. In 17 and 18, this was a bug where the current directory was overriding the scope. We fixed it in version 19.

**Reply by simonhaenisch** [reply]
🙄Why were people confused about this behavior 😄 I found it very convenient that once I'm in my project's folder, everything is automatically scoped to that project and team.

I work with 3 different teams plus my personal scope, so switching was as tedious as `--scope`. Also it was hard to keep track of what my active scope was which is why I always stay in my personal scope now, and use the `--scope` flag for team projects.

Anyway, I agree that it makes sense that additionally `alias` uses the local project's scope 😊

**Reply by paulogdm** [reply]
Have you considered using https://vercel.com/docs/v2/custom-domains instead of `alias`?

**Reply by simonhaenisch** [reply]
Yes I have but in that team it would involve restructuring our branches (which is not so simple because it's a mono-repo for both backend and frontend), so we're currently deploying to our production url manually (i. e. from a local machine).
